modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2171,12 +2171,44 @@ def map(
         PandasDataframe
             A new dataframe.
         """
-        map_fn = (
-            self._partition_mgr_cls.lazy_map_partitions
-            if lazy
-            else self._partition_mgr_cls.map_partitions
-        )
-        new_partitions = map_fn(self._partitions, func, func_args, func_kwargs)
+        if self.num_parts <= 1.5 * CpuCount.get():
+            map_fn = (
+                self._partition_mgr_cls.lazy_map_partitions
+                if lazy
+                else self._partition_mgr_cls.map_partitions
+            )
+            # old way
+            new_partitions = map_fn(self._partitions, func, func_args, func_kwargs)
+        else:
+            axis = (
+                1
+                if abs(self._partitions.shape[0] - CpuCount.get())
+                < abs(self._partitions.shape[1] - CpuCount.get())
+                else 0
+            )
+            column_splits = (
+                self._partitions.shape[0]
+                // (CpuCount.get() // self._partitions.shape[1])
+                if CpuCount.get() > self._partitions.shape[1]
+                else 1
+            )
+
+            if axis == 1 or column_splits <= 1:
+                # splitting by full_axis rows
+                new_partitions = self._partition_mgr_cls.map_axis_partitions(
+                    axis,
+                    self._partitions,
+                    func,
+                    keep_partitioning=True,
+                    map_func_args=func_args,
+                    map_func_kwargs=func_kwargs,
+                )
+            else:
+                new_partitions = (
```

Comment:
Put a comment similar to the one in if branch

Reply:
Done